src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx

```diff
@@ -515,9 +515,10 @@ export const ProspectingPage: React.FC = (): JSX.Element => {
         // Hide the loading indicator
         formikHelpers.setSubmitting(false);
 
-        // In case this prospect already existed in the corpus database,
+        // In case this prospect already existed in the corpus database i.e it has an approvedItem on it,
         // mark it as curated at this point
-        if (approvedItem?.url === currentProspect?.url) {
+        // A manually added item/currentProspect won't have an approvedItem on it so this check is never hit
+        if (currentProspect?.approvedCorpusItem) {
```

Comment:
Originally, we weren't resolving an `approvedItem` on the `Prospect` type in the graph. So we had two state variables that were being set and their urls were being compared to see if they were the same item/if it already existed in the corpus DB.

This lead to manually added items to also pass this check but not have an `id` since the item didn't come through the ML job/prospect-api, hence the error was thrown for marking it as curated but the initial schedule mutation passed. 

Now we can simply check if the prospect has an `approvedItem` on it to verify that is in the Corpus DB already and mark it as curated when an user schedules it. Note we already use this logic below to render the `<ExistingProspectCard>` component.

Reply:
thank you for this explanation!